### DISCUSSION TITLE
Is it possible to change the Emitter's particle speed at runtime?

### DISCUSSION THREAD

**Discussion by OswaldTheMighty** [matteobruni/tsparticles#1530] [Q&A] [answered]
I'm trying to update the emission speed (`emitters.particles.move.speed`) of emitters in real time - is this a thing?
I'm very new to the code base so I'm not 100% sure what can and cannot be updated at runtime. I noticed you can iterate through the particles and change stuff on them (e.g. speed), but I couldn't find a similar way to do the same things for emitters.

I tried to hack away (using Angular + ng-particles) a bit, but this doesn't seems to be the correct way and doesn't work (unsurprisingly).
```
const emitters: any = this.particleContainer.plugins.get("emitters");
if (emitters) {
    const myEmitter = emitters.array[0];
    myEmitter.emitterOptions.particles.move.speed = moveSpeed;
}
```

Is there a proper way of doing this?

**Comment by matteobruni** [comment, marked as answer]
I noticed that I added an `add`, `play` and `pause` methods but not a `get`. I'm adding it to the upcoming `1.27.0` release.

About the options updates I'll search deeply since the code you posted should have worked.

**Reply by matteobruni** [reply]
Ok, I found it. If you use `particlesOptions` instead of `emitterOptions` it works, I duplicated them for animations, so `emitterOptions` are just for emitter related options like spawn count or delay, and `particlesOptions` for the spawned particles.

You just can replace `emitterOptions.particles` with `particlesOptions`

**Reply by OswaldTheMighty** [reply]
Hi Matteo, awesome, you're absolutely right, that did the trick! Thanks for the help
